alternatives.c

```diff
@@ -1304,13 +1328,18 @@ int main(int argc, const char **argv) {
     struct stat sb;
     struct linkSet newSet = {NULL, NULL, NULL};
 
+    if (canUseAlternativeAdminDirUnderOSTree()) {
+        stateDir = "/etc/alternatives.admindir";
+    }
+
     setlocale(LC_ALL, "");
     bindtextdomain("chkconfig", "/usr/share/locale");
     textdomain("chkconfig");
 
     if (!argv[1])
         return usage(2);
 
+    int statedir_changed = 0;
```

Comment:
There is no real need for a new variable.  I'd suggest initializing `stateDir` to NULL and check it later instead.

Reply:
Thanks!